packages/jsapi-components/src/RefreshTokenUtils.ts

```diff
@@ -0,0 +1,39 @@
+import { createContext } from 'react';
+import Cookies from 'js-cookie';
+
+export const REFRESH_TOKEN_KEY = 'io.deephaven.web.client.auth.refreshToken';
+
+export type RefreshToken = { bytes: string; expiry: number };
+
+export const RefreshTokenContext = createContext<RefreshToken | null>(null);
+
+/**
+ * Read the refresh token from our cookie store
+ * @returns RefreshToken if it exists, null otherwise
+ */
+export function readRefreshToken(): RefreshToken | null {
+  const cookieToken = Cookies.get(REFRESH_TOKEN_KEY);
+  return cookieToken != null ? JSON.parse(cookieToken) : null;
```

Comment:
This may throw if the token fails to parse as JSON (i.e. a non-JSON string). Have you checked what happens if that's the case? I'm guessing this _shouldn't_ happen since we're setting the cookie typically, but want to make sure it doesn't just break everything (or at least breaks in a somewhat reasonable way)

Reply:
Yea this would be a problem on Enterprise as well.

Agreed, shouldn't be a problem, but I should wrap in a try catch anyway.